src/dotty/tools/dotc/core/unpickleScala2/Scala2Unpickler.scala

```diff
@@ -106,7 +106,7 @@ object Scala2Unpickler {
         // `denot.sourceModule.exists` provision i859.scala crashes in the backend.
         denot.owner.thisType select denot.sourceModule
       else selfInfo
-    denot.info = ClassInfo(denot.owner.thisType, denot.classSymbol, Nil, decls, ost) // first rough info to avoid CyclicReferences
+    denot.info = new TempClassInfo(denot.owner.thisType, denot.classSymbol, decls, ost) // first rough info to avoid CyclicReferences
```

Comment:
It's not clear to me where the `suspensions` of this TempClassInfo is executed. For the TempClassInfo in Namer, there's an explicit call `tempClassInfo.suspensions.foreach(_())`.